packages/connect/src/core/index.ts

```diff
@@ -123,7 +127,7 @@ const findUiPromise = <T extends UiPromiseResponse['type']>(promiseEvent: T) =>
     _uiPromises.find(p => p.id === promiseEvent) as UiPromise<T> | undefined;
 
 const removeUiPromise = (promise: Deferred<any>) => {
-    _uiPromises = _uiPromises.filter(p => p !== promise);
+    _uiPromises.splice(0).push(..._uiPromises.filter(p => p !== promise));
```

Comment:
not sure I get it why, now:
1. clear _uiPromises arrray
2. push new items with filter applied

so the difference here should be that the array is not recreated. is it because of types or something?

Reply:
the difference is few lines above. i've changed `let` to `const` therefore _uiPromises array is cleared by `splice` not by reassignment

Reply:
at first i wanted to move whole _uiPromise related code from code to a separate file  (events/ui-promise) but then i noticed that it will require more changes here and there (as usual) so i decided to leave it here for now, maybe i will do it in next batch :)